rasterio/_base.pyx

```diff
@@ -851,7 +851,14 @@ cdef class DatasetBase(object):
             dm = dm.encode('utf-8')
             domain = dm
 
-        band = self.band(bidx)
+        if not bidx > 0 and ovr:
+            raise Exception("Band index (bidx) option needed for overview level")
```

Comment:
Note: need a better error

Reply:
I think `errors.RasterioError` would be fine.